cudapoa/src/application_parameters.cpp

```diff
@@ -0,0 +1,225 @@
+/*
+* Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+*
+* NVIDIA CORPORATION and its licensors retain all intellectual property
+* and proprietary rights in and to this software, related documentation
+* and any modifications thereto.  Any use, reproduction, disclosure or
+* distribution of this software and related documentation without an express
+* license agreement from NVIDIA CORPORATION is strictly prohibited.
+*/
+
+#include "application_parameters.hpp"
+#include <fstream>
+#include <getopt.h>
+#include <iostream>
+#include <string>
+
+#include <claraparabricks/genomeworks/io/fasta_parser.hpp>
+#include <claraparabricks/genomeworks/utils/signed_integer_utils.hpp>
+#include <claraparabricks/genomeworks/version.hpp>
+
+namespace claraparabricks
+{
+
+namespace genomeworks
+{
+
+namespace cudapoa
+{
+
+ApplicationParameters::ApplicationParameters(int argc, char* argv[])
+{
+    struct option options[] = {
+        {"input", required_argument, 0, 'i'},
+        {"result", required_argument, 0, 'r'},
```

Comment:
Instead of result, let's make this an argument-less option to output "msa or not". by default we can output consensus

Reply:
This allows us to support mode 2 in the future, which can be to run both. Are we not going to have that mode later on?

Reply:
Indeed, and that's what the module API supports as well. But we haven't seen any use case yet where a user wants to run both, and I'm not sure if that's tested properly either ;). So let's make it more restricted for now, and open up later if we get a feature request.

Reply:
Sounds good. I've changed it to use the flag `-a` since the rest were already taken